### ISSUE TITLE
Closing Heredoc Indentation

### ISSUE THREAD

**Issue by siegfault** [rubocop-hq/rubocop#5406]
When autocorrecting heredoc indentation, the heredoc close is ignored. I think I would expect it to be realigned with the heredoc open.

The following examples use
```ruby
class Foo
  <<-SQL.strip_indent
    foo
SQL
end
```
as starting code.

--------

## Expected behavior

When running rubocop, I'd expect to see a violation for `SQL` needing to be indented two spaces and for the autocorrect to result in
```ruby
class Foo
  <<-SQL.strip_indent
    foo
  SQL
end
```

## Actual behavior

Instead, there are no violations.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.1 x86_64-linux)
```

## Further Discussion

I'm not sure whether this would fall on the IndentHeredoc rule (which currently only handles the innards of the heredoc), the IndentationWidth rule (which handles the opening), a new ClosingHeredocIndentation rule, or somewhere else. I'd be happy to work on this if given some indication of a preferred location.

**Comment by siggymcfried** [comment]
Any thoughts here?

**Comment by Drenmi** [comment]
Either `IndentationWidth` or `ClosingHeredocIndentation` seem like a good place. 🙂

**Comment by siggymcfried** [comment]
Sounds good - thanks! I'll see if I can take a look at this this weekend.

**Comment by Drenmi** [comment]
Swell! Feel free to ping me here if you have any questions.

**Comment by siggymcfried** [comment]
@Drenmi I opened https://github.com/bbatsov/rubocop/pull/5557 and would love comments! I started with `IndentationWidth` since I thought that might be preferable, but got a bit stumped trying to make it work there. Then I got `ClosingHeredocIndenation` working. Knowing what I know now, I might revisit an `IndentationWidth` implementation, but that will probably have to wait until the weekend.

**Comment by Drenmi** [comment]
> I opened #5557 and would love comments!

Will have a look!

> I started with IndentationWidth since I thought that might be preferable, but got a bit stumped trying to make it work there.

Expected so. It's already an insanely complex thing. 🙂
